waspc/src/Wasp/Generator/ServerGenerator/AuthG.hs

```diff
@@ -105,3 +133,80 @@ genMeRoute auth = return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplD
       object
         [ "userEntityLower" .= (Util.toLowerFirst (AS.refName $ AS.Auth.userEntity auth) :: String)
         ]
+
+genPassportAuth :: AS.Auth.Auth -> Generator [FileDraft]
+genPassportAuth auth =
+  sequence
+    [ genPassportJs auth
+    ]
+    <++> (if AS.Auth.isGoogleAuthEnabled auth then genGoogleAuth auth else return [])
```

Comment:
Same comment as above, seems like we usually do the "generate this only if needed" check inside the function (in this case it would be inside `genGoogleAuth`).